### PR TITLE
Use multistage build

### PR DESCRIPTION
**Pull request by sbrunner**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,40 @@
-FROM ubuntu:18.04
+FROM ubuntu:18.04 as builder
+LABEL maintainer="info@camptocamp.com"
+
+RUN apt-get -y update && \
+    LC_ALL=C DEBIAN_FRONTEND=noninteractive apt-get install -y bison flex python-lxml libfribidi-dev swig cmake librsvg2-dev colordiff libpq-dev libpng-dev libjpeg-dev libgif-dev libgeos-dev libgd-dev libfreetype6-dev libfcgi-dev libcurl4-gnutls-dev libcairo2-dev libgdal-dev libproj-dev libxml2-dev libxslt1-dev python-dev php-dev libexempi-dev lcov lftp libgdal-dev ninja-build git curl ccache clang libprotobuf-c-dev protobuf-c-compiler libharfbuzz-dev libcairo2-dev librsvg2-dev
```
Comment (danduk82) [inline Dockerfile]: idem here we could clean the apt environment

File: Makefile [hunk 2]
```diff
@@ -23,49 +23,29 @@ endif
 
 all: acceptance
 
-.PHONY: build acceptance build_acceptance_config build_acceptance
-
+.PHONY: pull
 pull:
 	for image in `find -name Dockerfile | xargs grep --no-filename FROM | awk '{print $$2}'`; do docker pull $$image; done
```
Comment (danduk82) [inline Makefile]: `grep '^FROM'` would be slightly safer

**Comment by sbrunner** [comment]
@danduk82 Done :-)

**Comment by danduk82** [comment]
but it seems that the tests are not happy

**Comment by sbrunner** [comment]
I suspect an error in mapserver :-(

**Comment by sbrunner** [comment]
=> https://github.com/mapserver/mapserver/pull/5554#issuecomment-589554349

File: Dockerfile [hunk 3]
```diff
@@ -1,4 +1,46 @@
...
+ARG MAPSERVER_BRANCH
+
+RUN git clone https://github.com/mapserver/mapserver --branch=${MAPSERVER_BRANCH} --depth=100 /src
+
+COPY checkout_release /tmp
+RUN cd /src; /tmp/checkout_release ${MAPSERVER_BRANCH}
+
+ENV \
+    CXX="/usr/lib/ccache/clang++" \
+    CC="/usr/lib/ccache/clang"
```
Comment (pvalsecc) [inline Dockerfile]: I'd say that ccache is useless in a multi-stage build context. You should remove it everywhere...

Comment (sbrunner) [reply Dockerfile]: +1, thanks :-)

**Pull request merged**